### ISSUE TITLE
I need a multi-touch option for touch environments.

### ISSUE THREAD

**Issue by Angsimosaurus** [translate-tools/linguist#511]
All the features work great on PC, but trying them out on Android felt like a lengthy process: 'hamburger menu -> Linguist -> Translate page & auto-translate current site'. It would be nice to have an option for touch environments, such as displaying the translation options with a three-finger tap or translating the current page directly.

**Comment by vitonsky** [comment]
It looks as duplicate for #357

Please explain your idea with three-finger tap. How it would may work

**Comment by Angsimosaurus** [comment]
> It looks as duplicate for #357
> 
> Please explain your idea with three-finger tap. How it would may work

For example, touching the screen with three fingers simultaneously could bring up the pop-up that usually appears when clicking the extension icon, or maybe even directly translate the current webpage.

In the case of the '[Translate Web Pages](https://github.com/FilipePS/Traduzir-paginas-web)' extension, when you touch the screen with three fingers, a pop-up appears asking if you want to translate the current webpage. The '[Immersive translate](https://immersivetranslate.com/en/)' extension, on the other hand, has a feature where it translates the webpage when you touch it with three fingers.

**Comment by vitonsky** [comment]
@Angsimosaurus sounds absolutely disgusting fro me as for user, since this behaviour will affect on my interaction with site, and may destroy my experience in case site have some features with three finger gestures.

This problem is common for low quality translators like Immersive translate who interfere in page content too much

**Comment by Angsimosaurus** [comment]
> @Angsimosaurus sounds absolutely disgusting for me as for user, since this behaviour will affect on my interaction with site, and may destroy my experience in case site have some features with three finger gestures.
> 
> This problem is common for low quality translators like Immersive translate who interfere in page content too much

Do you also find the popup approach like TWP undesirable?

**Comment by vitonsky** [comment]
@Angsimosaurus No, i think that's fine. I see your concern and totally agreed that current behaviour must be improved.
Currently i prefer the solution with a float panel on top/down of screen that will have most used controls. This solution looks most minimal invasive and suitable for most users. Is this approach used in TWP too, right?

But i agree that feature proposed by you, may be opt-in feature for users who used to this way. The feature would be disabled by default, and can be enabled manually in settings if user sure it will works fine for their use cases.

Let's talk about details. I have next questions to you, to understand the ideal implementation.

- What exactly features you need on mobile device to be available in quick access?
- What features could be placed in floating popup at bottom of viewport?
- What exactly should happens when user touch the screen with 3 fingers? You can also record and demonstrate here current behaviour in other extensions, that is close to ideal by your opinion.

**Comment by Angsimosaurus** [comment]
> @Angsimosaurus No, i think that's fine. I see your concern and totally agreed that current behaviour must be improved. Currently i prefer the solution with a float panel on top/down of screen that will have most used controls. This solution looks most minimal invasive and suitable for most users. Is this approach used in TWP too, right?
> 
> But i agree that feature proposed by you, may be opt-in feature for users who used to this way. The feature would be disabled by default, and can be enabled manually in settings if user sure it will works fine for their use cases.
> 
> Let's talk about details. I have next questions to you, to understand the ideal implementation.
> 
> * What exactly features you need on mobile device to be available in quick access?
> * What features could be placed in floating popup at bottom of viewport?
> * What exactly should happens when user touch the screen with 3 fingers? You can also record and demonstrate here current behaviour in other extensions, that is close to ideal by your opinion.

1. A simple and quick way to translate the current webpage, or automatically translating the domain without multiple taps is required for mobile devices.
2. The popup menu should include an option to choose whether you want to auto-translate the current domain along with the webpage translation functionality.
3. TWP allows you to display an elliptical popup at the top or bottom (position changeable) by touching three fingers simultaneously, just like in the Chrome browser for mobile. Pressing the button on the far right instantly translates the current page, and pressing the gear button gives options such as 'Always translate this language', 'Don't translate this language', 'Always translate this domain' and 'Don't translate this domain'.

![image](https://github.com/user-attachments/assets/47033640-6f9b-431d-9908-6ec73d9db27a)
